Google.Api.Generator/Generation/MethodDetails.cs

```diff
@@ -269,6 +269,12 @@ private static MethodDetails DetectPagination(ServiceDetails svc, MethodDescript
                 return null;
             }
 
+            if (pageSizeCandidate.Name == "max_results" && !(pageSizeCandidate.MessageType is null) && pageSizeCandidate.MessageType.FullName == "google.protobuf.UInt32Value")
```

Comment:
This is fine, but a simpler option would be:

```csharp
if (pageSizeCandidate.Name == "max_results" && pageSizeCandidate.MessageType?.FullName == "google.protobuf.UInt32Value")
```

Reply:
:+1: